keel-artifact/src/main/kotlin/com/netflix/spinnaker/keel/artifact/ArtifactListener.kt

```diff
@@ -92,37 +108,40 @@ class ArtifactListener(
    */
   // todo eb: should we fetch more than one version?
   @Scheduled(fixedDelayString = "\${keel.artifact-refresh.frequency:PT6H}")
-  fun syncArtifactVersions() =
-    runBlocking {
-      repository.getAllArtifacts().forEach { artifact ->
-        launch {
-          val lastRecordedVersion = getLatestStoredVersion(artifact)
-          val latestVersion = when (artifact) {
-            is DebianArtifact -> getLatestDeb(artifact)?.let { "${artifact.name}-$it" }
-            is DockerArtifact -> getLatestDockerTag(artifact)
-          }
-          if (latestVersion != null) {
-            val hasNew = when {
-              lastRecordedVersion == null -> true
-              latestVersion != lastRecordedVersion -> {
-                artifact.versioningStrategy.comparator.compare(lastRecordedVersion, latestVersion) > 0
-              }
-              else -> false
+  fun syncArtifactVersions() {
+    if (enabled.get()) {
```

Comment:
I noticed some log traffic related to this scheduled method running when Spring tests were executing so I made it discovery-dependent like the other scheduled things

Reply:
nice